snap/info.go

```diff
@@ -409,6 +409,12 @@ type SlotInfo struct {
 	Apps      map[string]*AppInfo
 }
 
+// SocketInfo provides information on application sockets.
+type SocketInfo struct {
+	ListenStream string
```

Comment:
The "listen-stream" option must be documented and implemented as supporting both ints and strings, so people can use 8080 as in systemd without being surprised that this breaks down. For Go, there's no need to handle it specially as it will transparently unmarshal the string properly. Python (snapcraft) may need a conditional to handle both cases properly, though. Let's please have a small test ensuring ints work too.